test/k8sT/DatapathConfiguration.go

```diff
@@ -308,10 +308,14 @@ var _ = Describe("K8sDatapathConfig", func() {
 		})
 
 		It("Check connectivity with automatic direct nodes routes", func() {
-			deploymentManager.DeployCilium(map[string]string{
+			options := map[string]string{
 				"global.tunnel":               "disabled",
 				"global.autoDirectNodeRoutes": "true",
-			}, DeployCiliumOptionsAndDNS)
+			}
+			if helpers.RunsWithKubeProxy() {
```

Comment:
Maybe a small comment explaining why (1) when we run w/ kube-proxy, (2) we disable masquerade for this particular test case.

Reply:
@brb TBH, I'm not sure I understand why that is necessary. It should work with masquerade enabled, no? Is there an issue opened somewhere to track this?